src/Dialect/ONNX/Rewrite.td

```diff
@@ -952,4 +952,78 @@ def SoftmaxV11ToLatestPattern: Pat<
    [(AxisIsTheLastDim $input, $axisAttr)]
 >;
 
+//===----------------------------------------------------------------------===//
```

Comment:
Is there any chance that would could add this decomposition in its own pass or as an opt in decomposition in the ONNX decomposition pass.

`Rewrite.td` is canonicalization. I'm not sure if its correct to consider the decomposition of `ONNXConvTranspose` into `ONNXConvOps` as canonicalization.

However I'm sure if we looked at `rewrite.td` we would see other passes which also maybe should exist in a different pass.

Reply:
@philass 
It seems the ONNX decomposition pass is appropriate for this kinds of rewriting, but I implemented this rewriting here because I wanted to do it after shape inference. 
I will consider whether the decomposition pass or new different pass is better. Thanks!

Reply:
I created new pass called DecomposeONNXToONNXAfterPass and move the rewriting into it. This pass is for decomposing onnx operation after shape inference.

Reply:
I found this decomposition of this PR can be included in original decompose pass (DecomposeONNXToONNXPass) by adding ConvTranspose op as dynamic legality op as @tungld suggested below.